src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -22,6 +22,16 @@ def _resolve_aliases(target):
     else:
       yield declared
 
+    for export in _resolve_exports(declared):
+      yield export
+
+
+def _resolve_exports(target):
+  for export in getattr(target, 'exports', []):
+    for exp in _resolve_exports(export):
```

Comment:
Hm. I feel like they shouldn't be transitive. If the goal is to make the exports of a library explicit at the library's declaration, then I think exports being transitive is confusing.

For example, in order to find all the exported targets of `t(exports=['a','b','c'])`, you'd need to look at `a`, `b` and `c`'s declarations too, and their exported targets declarations if they exist. If exports are intransitive, then all the necessary information is in `t(exports=['a','b','c'])`.